compiler/aarch64/codegen/ControlFlowEvaluator.cpp

```diff
@@ -480,7 +480,81 @@ TR::Register *
 OMR::ARM64::TreeEvaluator::tableEvaluator(TR::Node *node, TR::CodeGenerator *cg)
 	{
 	// TODO:ARM64: Enable TR::TreeEvaluator::tableEvaluator in compiler/aarch64/codegen/TreeEvaluatorTable.hpp when Implemented.
-	return OMR::ARM64::TreeEvaluator::unImpOpEvaluator(node, cg);
+   int32_t       total = node->getNumChildren();
+   int32_t       numBranchTableEntries = total - 2;
+   TR::Register   *selectorReg           = cg->evaluate(node->getFirstChild());
+   int32_t       i;
+   TR::Node       *secondChild = node->getSecondChild();
+   TR::RegisterDependencyConditions *conditions;
+   TR::Register   *t1Register;
+
+   if (numBranchTableEntries >= 5)
+      {
+      // Note: When implementation is changed with regards to number of used registers (or dependency),
+      // the matching GlDepRegs free checks in OMRCodeGenerator.cpp#getMaximumNumberOfGPRsAllowedAcrossEdge(..) must be updated to synch.
+      conditions  = new (cg->trHeapMemory()) TR::RegisterDependencyConditions(2, 2, cg->trMemory());
+      t1Register = cg->allocateRegister();
+      addDependency(conditions, t1Register, TR::RealRegister::NoReg, TR_GPR, cg);
+      }
+   else
+      {
+      conditions = new (cg->trHeapMemory()) TR::RegisterDependencyConditions(1, 1, cg->trMemory());
+      }
+
+   addDependency(conditions, selectorReg, TR::RealRegister::NoReg, TR_GPR, cg);
+
+   if (secondChild->getNumChildren() > 0)
+      {
+      cg->evaluate(secondChild->getFirstChild());
+      conditions = conditions->clone(cg, generateRegisterDependencyConditions(cg, secondChild->getFirstChild(), 0));
+      }
+
+   if (numBranchTableEntries < 5)
+      {
+      for (i = 0; i < numBranchTableEntries; i++)
+         {
+         generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::ccmpimmw, node, t1Register, selectorReg, i, 0);
+         generateConditionalBranchInstruction(cg, TR::InstOpCode:: b_cond, node, node->getChild(2+i)->getBranchDestination()->getNode()->getLabel(), TR::CC_EQ);
+         }
+      generateConditionalBranchInstruction(cg, TR::InstOpCode:: b_cond, node, secondChild->getBranchDestination()->getNode()->getLabel(), TR::CC_AL, conditions);
+      }
+   else
+      {
+      TR::RealRegister *machineIP        = cg->machine()->getRealRegister(TR::RealRegister::gr31);
+      uint32_t      base, rotate;
+
+      if (!constantIsImmed8r(numBranchTableEntries, &base, &rotate))
```

Comment:
This is the wrong function to call if you're using a CCMP immediate instruction (which only supports a 5-bit immediate).  Please read the description in section 6.2.45 of the ARM architecture manual.